fix(icons): use deprecated + non-standard icons consistently

danielhjacobs opened the pull request:
## Summary

Fixes number 1 and 2 (edit: 1-3) from #5571 (apologies if the issue author wanted to create this PR...)

### Problem

Non-standard / deprecated icons are not consistent

### Solution

Be consistent with non-standard / deprecated icons. I am not a designer, so I copy-pasted https://github.com/mdn/yari/edit/main/client/src/assets/icons/note-warning.svg for badges.svg, using an appropriate transform to scale. If someone can suggest a better [d](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d) to avoid the transform, please do so.

---

## Screenshots

### Before

![Image](https://camo.githubusercontent.com/53fd1b4e8942f7836611131e5ec2b1e7048c6cb58bb68c4080e00562cc408ac1/68747470733a2f2f692e696d6775722e636f6d2f3456734a6f41642e706e67)

![Image2](https://camo.githubusercontent.com/669350ad59cda7a3ea9e3f399a3fb02a3161f14c66240766d042213c6c59ceae/68747470733a2f2f692e696d6775722e636f6d2f7a6b49635875572e706e67)

![Image3](https://camo.githubusercontent.com/09c517afabec77e333c173b02e483e6778c05e546e9fcd1c441f382664a1a60d/68747470733a2f2f692e696d6775722e636f6d2f4c4d6965616f6e2e706e672535422f696d67)

### After

![Screenshot from 2022-03-11 15-07-13](https://user-images.githubusercontent.com/29206584/157952867-ca885e2e-d6e2-4882-ae41-5ee44a0824f8.png)

![Screenshot from 2022-03-11 15-12-19](https://user-images.githubusercontent.com/29206584/157953388-7b7c8cca-7200-49d8-b28f-3e05a7d9366a.png)

![Screenshot from 2022-03-11 15-13-36](https://user-images.githubusercontent.com/29206584/157954227-7f0c99e2-76e0-495b-a134-e9d9b6a431e7.png)

OnkarRuikar commented:
It's ok.

There are some more places in CSS code where deprecated is associated with thumbs down. We need to consider them as well:

client\src\document\ingredients\browser-compatibility-table\atoms\_bc-icons.scss
```css
.ic-deprecated {
  &::before {
    background-image: url("~@mdn/dinocons/emojis/thumbs-down.svg");
    content: "";
  }
}
```

client\src\document\molecules\_notecards.scss
```css
&.deprecated {
    @include set-notecard-icon-only("~@mdn/dinocons/emojis/thumbs-down.svg");
}
```

I don't know where exactly these CSS rules reflect on the UI.

***
On page https://developer.mozilla.org/en-US/docs/Web/API/Window/cancelIdleCallback
The experimental notecard has info icon. It would be great if we make it flask.
The html code already marks the element with `experimental` class. 
```html
<div class="notecard experimental" id="sect1">
  <p>
      <strong>Experimental:</strong> 
       ...
  </p>
</div>
```
We need to add `.notecard.experimental` CSS rule. Like you did for `.notecard.deprecated` in client/src/ui/molecules/notecards/index.scss

danielhjacobs commented:
I addressed the feedback in my second commit. I also couldn't find where in the UI related to those CSS rules, but I used what seems to be the proper path based on the location of the trash svg in https://github.com/mdn/mdn-dinocons.

My change to client/src/ui/molecules/notecards/index.scss seems to have worked to switch the experimental icon to the flask.

caugner commented:
To change the icon of the non-standard note, we will need to add a CSS class to the notecards generated by `kumascript/macros/Non-standardGeneric.ejs`, similar to how `kumascript/macros/DeprecatedGeneric.ejs` does this.

caugner commented:
> Isn't the non-standard icon already working

You're right, it's basically a side-effect of non-standard and warning having the same icon.

What I would suggest is to use a dedicated `non-standard` CSS class instead of `warning` in `Non-standardGeneric.ejs`, just to avoid that changing the `warning` icon in the future causes the `non-standard` icon to longer match.

If you think that's not a good idea, I will go ahead and merge your PR as is.

danielhjacobs commented:
> use a dedicated `non-standard` CSS class instead of `warning` in Non-standardGeneric.ejs, just to avoid that changing the `warning` icon in the future causes the `non-standard` icon to longer match.

Done, although the same icon is referenced. I can duplicate https://github.com/mdn/yari/blob/main/client/src/assets/icons/note-warning.svg into https://github.com/mdn/yari/blob/main/client/src/assets/icons/non-standard.svg. Should I do so?

danielhjacobs commented:
I don't actually think the note-warning class is used for anything but non-standard support though. note-info is used for actual notes. (the circle with the exclamation point).

danielhjacobs commented:
I left the note-warning CSS class and SVG despite my assumption that they are not used anymore just in case I'm mistaken.

caugner commented on client/src/document/ingredients/browser-compatibility-table/legend.tsx:
```diff
@@ -11,8 +11,8 @@ const LEGEND_LABELS = {
   experimental: "Experimental. Expect behavior to change in the future.",
-  "note-warning": "Non-standard. Check cross-browser support before using.",
-  "thumbs-down": "Deprecated. Not for use in new websites.",
+  "non-standard": "Non-standard. Check cross-browser support before using.",
```
I had a quick look and this should be `nonstandard` without hyphen actually, to be consistent:

```suggestion
  nonstandard: "Non-standard. Check cross-browser support before using.",
```

caugner commented on client/src/ui/atoms/icon/index.scss:
```diff
@@ -2,9 +2,9 @@ $icons: "add-filled", "add", "altname", "bell-filled", "bell", "bookmark-filled"
-  "menu-filled", "menu", "mobile", "moon-fill", "next", "no", "note-info",
-  "note-warning", "note-deprecated", "partial", "prefix", "preview", "previous",
-  "quote", "search", "server", "sidebar", "small-arrow", "sun-fill",
+  "menu-filled", "menu", "mobile", "moon-fill", "next", "no", "non-standard",
```
```suggestion
  "menu-filled", "menu", "mobile", "moon-fill", "next", "no", "nonstandard",
```

caugner commented on client/src/ui/molecules/notecards/index.scss:
```diff
@@ -53,7 +53,15 @@
   }
 
-  &.deprecated,
+  &.non-standard {
```
```suggestion
  &.nonstandard {
```

caugner commented on client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:
```diff
@@ -76,12 +76,12 @@ function StatusIcons({ status }: { status: bcd.StatusBlock }) {
       title: "Non-standard. Expect poor cross-browser support.",
       text: "Non-standard",
-      iconClassName: "icon-note-warning",
+      iconClassName: "icon-non-standard",
```
```suggestion
      iconClassName: "icon-nonstandard",
```

caugner reviewed (changes requested):
Sorry, I added some remarks all over the place, but I hope you don't mind.

- When I recommended `non-standard`, I didn't consider that other places already use this class as `nonstandard`, so to be consistent, we should go hyphen-less here.
- Wherever we already have semantical icon names (like `icon-preview`) instead of descriptive names (like `icon-flask`), we should be consistent (i.e. deprecated/nonstandard instead of trash/thumbs-down).

Once this is done, everything should be tidier than before.

danielhjacobs commented on client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:
```diff
@@ -76,12 +76,12 @@ function StatusIcons({ status }: { status: bcd.StatusBlock }) {
       title: "Deprecated. Not for use in new websites.",
       text: "Deprecated",
-      iconClassName: "icon-thumbs-down",
+      iconClassName: "icon-trash",
```
Tried adding it. In doing so, found the root cause of #5588.

It overrides badges.svg, which has better icons for the sidebar.

danielhjacobs commented on client/src/document/molecules/_notecards.scss:
```diff
@@ -156,8 +156,4 @@
     color: $mdn-color-dark-grey-90;
   }
-
-  &.deprecated {
```
Not sure, but this should be covered by https://github.com/mdn/yari/pull/5583/files#diff-40d0c7f6caf8612b5ec3889b21a9df046639377d9212766627f8eccc574868a1

danielhjacobs commented:
Addressed all feedback.

danielhjacobs commented:
Pictures for reassurances:

Changed pages:
![Screenshot from 2022-03-14 15-22-17](https://user-images.githubusercontent.com/29206584/158245633-b7714aa2-8777-4af9-8ec4-3d073f6b44f1.png)
![Screenshot from 2022-03-14 15-22-10](https://user-images.githubusercontent.com/29206584/158245638-80019dbb-4a45-401b-b8b8-aa4aef0907e4.png)
![Screenshot from 2022-03-14 15-21-44](https://user-images.githubusercontent.com/29206584/158245640-f7c88483-13f4-401d-bb6e-f52964cf5bba.png)
![Screenshot from 2022-03-14 15-22-24](https://user-images.githubusercontent.com/29206584/158245631-df7266a7-514d-455f-b8f4-c6e329ee5a9d.png)

danielhjacobs commented:
Sorry, I made a very minor change. One more reference to trash was left over,

caugner commented on client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss:
```diff
@@ -62,7 +62,7 @@
 
 .ic-deprecated {
```
@danielhjacobs Did you see by any chance where these `ic-*` CSS classes are being used?

danielhjacobs replied:
I assume they're not. Perhaps they're from pre-redesign.

```
[... yari]$ grep -rl --exclude-dir=node_modules 'ic-deprecated' .
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
```

caugner replied:
Yup, it looks like it, which means we should be able to remove those `ic-*` classes from `_bc-icons.scss`.

<img width="390" alt="Screenshot 2022-03-14 at 20 44 09" src="https://user-images.githubusercontent.com/495429/158248980-9c2bf8c7-3543-4385-a687-ef14542953e9.png">

danielhjacobs replied:
All of them or just those three?

danielhjacobs replied:
At a glance, the answer is all of them:

```
[... yari]$ grep -rl 'ic-footnote' .
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
[... yari]$ grep -rl 'ic-altname' .
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
[... yari]$ grep -rl 'ic-disabled' .
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
[... yari]$ grep -rl 'ic-prefix' .
./node_modules/axe-core/axe.js
./node_modules/axe-core/axe.min.js
./node_modules/axe-core/lib/commons/text/is-valid-autocomplete.js
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
```

The other results for ic-prefix are horrific-prefix.

caugner commented:
Awesome @danielhjacobs, thank you for your great work and patience! 🎉
